test/unit/spec/DeviceFingerprint_spec.js

```diff
@@ -102,23 +108,30 @@ function (Expect, $sandbox, DeviceFingerprint) {
         })
         .fail(function (reason) {
           expect(reason).toBe('device fingerprint is not supported on Windows phones');
+          var $iFrame = $sandbox.find('iframe');
+          expect($iFrame).not.toExist();
           done();
         });
     });
 
     it('ignores if message is not from right iframe', function (done) {
+      spyOn(DeviceFingerprint, 'isMessageFromCorrectSource').and.callFake(() => {
+        return false;
+      });
       mockIFrameMessages(true);
-      DeviceFingerprint.generateDeviceFingerprint(baseUrl, $sandbox)
-        .then(function () {
-          done.fail('Fingerprint promise should not have been resolved');
-        })
-        .fail(function (reason) {
-          done.fail('Fingerprint promise should not have been rejected. ' +  reason);
-        });
-      Expect.tick().then(function () {
-        // give it time to check if promise resolves or rejects.
+
+      DeviceFingerprint.generateDeviceFingerprint(baseUrl, $sandbox);
+      expect(DeviceFingerprint.isMessageFromCorrectSource).not.toHaveBeenCalled();
+
+      window.addEventListener('message', () => {
```

Comment:
Is this running in jest or karma? If karma, I think we'll need to remove this event listener to prevent interference with other tests